### PR TITLE
GitHub Enterprise Support 2: Electric Boogaloo

### LINKED ISSUE #208: Missing option to pass own GitHubAPI object.
Self-hosted GitHub instances have different hostname than `github.com`.
https://github.com/JuliaWeb/GitHub.jl provides `GitHubAPI` abstract type, which is first parameter to all it's calls. 
For self-hosted instance I need to pass it everywhere, now the default instance is used.
Should I make a PR for this?

### PR DESCRIPTION
**Pull request by racinmat**
Fixes #208 
Closes #221 

- Passing github base url as parameter.
- Removing hardcoded github.com from some places.

### CODE REVIEW

**Comment by racinmat** [comment]
@DilumAluthge  here it ts.

**Comment by DilumAluthge** [comment]
bors merge

File: src/new_versions.jl [hunk 1]
```diff
@@ -261,8 +274,8 @@ function make_pr_for_new_version(precommit_hook::Function,
     if new_pr_title in pr_titles
         @info("An open PR with the title already exists", new_pr_title)
     else
-        url_with_auth = "https://x-access-token:$(auth.token)@github.com/$(repo.full_name).git"
-        url_for_ssh = "git@github.com:$(repo.full_name).git"
+        url_with_auth = "https://x-access-token:$(auth.token)@$(api.endpoint.host)/$(repo.full_name).git"
+        url_for_ssh = "git@$(api.endpoint.host):$(repo.full_name).git"
```
[the remark this answers is not in the document]
Comment (DilumAluthge) [reply src/new_versions.jl:278]: Yeah. I'll work on fixing that stuff when I have a chance.

[the remark this answers is not in the document]
Comment (racinmat) [reply src/new_versions.jl:278]: I can fix it.

[the remark this answers is not in the document]
Comment (racinmat) [reply src/new_versions.jl]: Done in the last commit. I think these are all x-access-token usages, so I hope that's everything.

**Comment by DilumAluthge** [comment]
bors try

**Comment by DilumAluthge** [comment]
bors try

**Comment by DilumAluthge** [comment]
bors try

**Comment by DilumAluthge** [comment]
bors merge

**Pull request merged**